visualization/app/codeCharta/util/fileNameHelper.ts

```diff
@@ -13,23 +15,22 @@ export class FileNameHelper {
 	}
 
 	private static getFileNameWithoutTimestamp(fileName: string, isDeltaState: boolean) {
-		const dateRegex = /_\d{4}(?:-\d{1,2}){2}_\d{1,2}-\d{1,2}\./
-
 		if (!isDeltaState) {
-			if (dateRegex.test(fileName)) {
-				return fileName.slice(0, Math.max(0, dateRegex.exec(fileName).index))
+			const dateMatch = dateRegex.exec(fileName)
+			if (dateMatch) {
+				return fileName.slice(0, dateMatch.index)
 			}
-			if (fileName.includes(CodeChartaService.CC_FILE_EXTENSION)) {
-				return fileName.slice(0, Math.max(0, fileName.search(CodeChartaService.CC_FILE_EXTENSION)))
+			if (fileName.endsWith(CodeChartaService.CC_FILE_EXTENSION)) {
+				return fileName.slice(0, -CodeChartaService.CC_FILE_EXTENSION.length)
 			}
-			if (fileName.includes(FileNameHelper.JSON_EXTENSION)) {
-				return fileName.slice(0, Math.max(0, fileName.search(FileNameHelper.JSON_EXTENSION)))
+			if (fileName.endsWith(FileNameHelper.JSON_EXTENSION)) {
+				return fileName.slice(0, -FileNameHelper.JSON_EXTENSION.length)
 			}
 		}
 		return fileName
 	}
 
 	static withoutCCJsonExtension(fileName: string) {
-		return fileName.replace(".cc", "").replace(this.JSON_EXTENSION, "")
+		return fileName.replace(/(\.cc)?\.json?$/, "")
```

Comment:
Bug fix: only replace the `.cc` and `.json` at the end of the filename, not anywhere in-between.